### ISSUE TITLE
[@typescript-eslint/func-call-spacing] False positive on function type

### ISSUE THREAD

**Issue by pistone** [typescript-eslint/typescript-eslint#2978]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
See the code below.

.eslintrc.js:
```JS
module.exports = {
  'env': {
    'browser': true,
    'es2021': true,
  },
  'extends': [
    'google',
  ],
  'parser': '@typescript-eslint/parser',
  'parserOptions': {
    'ecmaVersion': 12,
    'sourceType': 'module',
  },
  'plugins': [
    '@typescript-eslint',
  ],
  'rules': {
    'func-call-spacing': "off",
    '@typescript-eslint/func-call-spacing': ["error"],
  },
};

```

```TS
// your repro code case
const x = new Map<number, (y:number)=>number> ();

```

**Expected Result**
No error is reported.

**Actual Result**
```
  1:45  error  Unexpected whitespace between function name and paren  @typescript-eslint/func-call-spacing
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.1` |
| `@typescript-eslint/parser`        | `4.14.1` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `7.18.0` |
| `node`                             | `14.15.1` |

**Comment by bradzacher** [comment]
```
const x = new Map<number, (y:number)=>number> ();
                                             ^ space here
```

I'm not sure I understand.
It looks like this is working correctly?

**Comment by pistone** [comment]
Yes, you are right. I was looking at the wrong column.
